packages/core/src/internal/base/button.base.ts

```diff
@@ -104,26 +119,48 @@ export class CdsBaseButton extends LitElement {
     if (!this.readonly) {
       if (this.disabled) {
         stopEvent(event);
-      } else if (!event.defaultPrevented) {
-        const buttonTemplate = html`<button
-          aria-hidden="true"
-          ?disabled="${this.disabled}"
-          tabindex="-1"
-          style="display: none !important"
-          value="${ifDefined(this.value)}"
-          name="${ifDefined(this.name)}"
-          type="${ifDefined(this.type)}"
-        ></button>`;
-
-        const marker = document.createElement('div');
-        this.appendChild(marker);
-        render(buttonTemplate, marker);
+      } else if (!event.defaultPrevented && this.marker && this.explicitClick) {
+        if (this.markerAttached()) {
```

Comment:
So we are only appending a marker if a marker is already attached? Maybe the language here is confusing me. But it seems to me like we would only append a marker if a marker wasn't attached...

Reply:
Very good question. Looking at it now, it seems pointless the way it is. But I first need to check why it is passing tests and make test changes if necessary.

Reply:
You are right here. Logically it has to be `if(!markerAttached())` 
And it works without having a native clickable button just because click can be attached to anything.
So, apart from the form-submit case, I am not really sure why else we need the native button.
N